src/Illuminate/Foundation/Exceptions/Handler.php

```diff
@@ -220,12 +220,20 @@ protected function prepareJsonResponse($request, Exception $e)
 
         $headers = $this->isHttpException($e) ? $e->getHeaders() : [];
 
-        return response(json_encode([
-            'message' => $e->getMessage(),
-            'file' => $e->getFile(),
-            'line' => $e->getLine(),
-            'trace' => $e->getTrace(),
-        ], JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES), $status, array_merge($headers, [
+        if (config('app.debug')) {
+            $response = [
+                'message' => $e->getMessage(),
+                'file' => $e->getFile(),
+                'line' => $e->getLine(),
+                'trace' => $e->getTrace(),
+            ];
+        } else {
+            $response = [
+                'message' => $this->isHttpException($e) ? $e->getMessage() : 'Server Error',
+            ];
+        }
+
+        return response(json_encode($response, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES), $status, array_merge($headers, [
```

Comment:
So we actually can only return JSON here, why not just:
```php
return new JsonResponse($response, $status, $headers, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES);
``` 
`@return` doblock should be changed accordingly